urwid/widget/columns.py

```diff
@@ -516,9 +671,15 @@ def column_widths(self, size: tuple[int] | tuple[int, int], focus: bool = False)
             if t == WHSettings.GIVEN:
                 static_w = width
             elif t == WHSettings.PACK:
-                # FIXME: should be able to pack with a different
-                # maxcol value
-                static_w = w.pack((maxcol,), focus and i == self.focus_position)[0]
+                w_sizing = w.sizing()
+                if Sizing.FLOW in w_sizing:
+                    # FIXME: should be able to pack with a different maxcol value
+                    w_size = (maxcol,)
+                elif Sizing.FIXED in w_sizing:
+                    w_size = ()
+                else:
+                    raise ColumnsError(f"Unusual widget sizing {w_sizing} for {t}")
```

Comment:
we can try to guess "FLOW", but in most cases we will have crazy long tracebacks with not human-friendly information here

Reply:
returned back "shoot in the leg" behavior with warning message and covered by test